osquery/logger/plugins/windows_event_log.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+
+#include "osquery/logger/plugins/windows_event_log.h"
+#include "osquery/logger/plugins/windows_event_log_manifest/osquery.h"
+
+namespace osquery {
+
+REGISTER(WindowsEventLoggerPlugin, "logger", "windows_event_log");
+
+WindowsEventLoggerPlugin::~WindowsEventLoggerPlugin() {
+  releaseHandle(registration_handle_);
+}
+
+Status WindowsEventLoggerPlugin::logString(const std::string& s) {
+  return emitLogRecord(registration_handle_, s);
+}
+
+Status WindowsEventLoggerPlugin::logStatus(
+    const std::vector<StatusLogLine>& log) {
+  for (const auto& item : log) {
+    auto status = emitLogRecord(registration_handle_,
+                                item.message,
+                                item.severity,
+                                item.filename,
+                                item.line);
+    if (!status.ok()) {
+      LOG(ERROR) << status.getMessage();
```

Comment:
perhaps this could `return Status(1, status.getMessage())` to bubble the issue up? It seems that right now this function is somewhat inconsistent with return values, and also will always return `Status(0, "OK");`